packages/components/src/DateTimePickers/InputDateTimePicker/InputDateTimePicker.component.js

```diff
@@ -198,28 +251,73 @@ class InputDateTimePicker extends React.Component {
 		this.unmountComponentContainerHandler();
 	}
 
-	onSubmitPicker({ date, time }) {
-		this.updateDateTime(date, time);
+	onSubmitPicker(event, { date, time }) {
+		const datetime = getObjectDate(date, time);
+		const errorMessage = undefined;
+
+		this.updateDatePartStateAndTriggerChange(event, {
+			date,
+			time,
+			textInput: getTextDate(date, time),
+			datetime,
+			errorMessage,
+		});
+
 		this.switchDropdownVisibility(false);
+		this.triggerBlur(event);
```

Comment:
Yes the final purpose is to trigger the `onFinish` event. But the idea was to have it the same way as for other component which is `onBlur`. I don't see why having a `blur` semantic in a component is weird ?
I trigger the `onBlur` on submit (not on selection) of the picker, because it's hidden in the same time, which remove the focus.

If I update it to be `onChange`, it will validate at each end of key stroke, which is not an identical behavior with others component like those with inputs like `Text`.